src/scheduler.cc

```diff
@@ -223,7 +224,7 @@ Status SchedulerService::TaskInfo(ServerContext* context, const TaskInfoRequest*
   }
   for (int i = 0; i < workers->size(); ++i) {
     OperationId operationid = (*workers)[i].current_task;
-    if (operationid != NO_OPERATION) {
+    if (operationid != NO_OPERATION && operationid != ROOT_OPERATION) {
```

Comment:
Since the code is going to be getting littered with this, can you make a function to do this check on the ID instead of checking the ID manually?